cli/src/main/java/com/devonfw/tools/ide/commandlet/CreateCommandlet.java

```diff
@@ -0,0 +1,75 @@
+package com.devonfw.tools.ide.commandlet;
+
+import com.devonfw.tools.ide.context.IdeContext;
+import com.devonfw.tools.ide.io.FileAccess;
+import com.devonfw.tools.ide.process.ProcessContext;
+import com.devonfw.tools.ide.process.ProcessResult;
+import com.devonfw.tools.ide.property.StringProperty;
+
+import java.nio.file.Path;
+
+/**
+ * {@link Commandlet} to create a new IDEasy instance
+ */
+public class CreateCommandlet extends Commandlet {
+
+  private final StringProperty newInstance;
+
+  /**
+   * The constructor.
+   *
+   * @param context the {@link IdeContext}.
+   */
+  public CreateCommandlet(IdeContext context) {
+
+    super(context);
+    addKeyword(getName());
+    newInstance = add(new StringProperty("", false, "newInstance"));
+  }
+
+  @Override
+  public String getName() {
+
+    return "create";
+  }
+
+  @Override
+  public void run() {
+
+    String newInstanceName = newInstance.getValue();
+    Path newInstancePath;
+
+    if (newInstanceName == null) {
+      newInstancePath = this.context.getCwd();
+    } else {
+      newInstancePath = this.context.getIdeRoot().resolve(newInstanceName);
+      this.context.getFileAccess().mkdirs(newInstancePath);
+    }
+
+    this.context.info("Creating new IDEasy instance in {}", newInstancePath);
+    if (!this.context.getFileAccess().isEmptyDir(newInstancePath)) {
+      this.context.askToContinue("Directory is not empty, continue?");
+    }
+
+    initializeInstance(newInstancePath);
+    ProcessContext pc = this.context.newProcess().executable("ide");
+    pc.addArgs("update");
+    pc.directory(newInstancePath);
+    if (pc.run() == ProcessResult.SUCCESS) {
+      this.context.success("IDEasy Instance successfully created in {}", newInstancePath);
+    } else {
+      this.context.warning("Could not create IDEasy Instance.");
+    }
```

Comment:
Do not call ourselves as process. If IDEasy is already launched what should we call another process that needs to boostrap and then do the things we already have in our Java code next door (next method)?
That is exactly why I suggested to create a super-class that both the `CreateCommandlet` and the `UpdateCommandlet` extend. Then simply add methods like `updateSettings` and `setupConf` there as protected methods and just call them from the sub-classes. If I am not mistaken, you could even orchestrate the entire process of the update (calling `updateSettings`, creating folders, `setupConf`, etc.) there in a single reusable method.
The only thing to consider is that you need to supply the path to IDE_HOME somehow or you would need to re-initialize the `IdeContext` on the fly.

Reply:
removed this block, still not implemented reinitializing of the context.